boards/BMS/Src/App/states/App_AllStates.c

```diff
@@ -157,6 +158,16 @@ bool App_AllStatesRunOnTick100Hz(struct StateMachine *const state_machine)
         status = false;
         App_SharedStateMachine_SetNextState(state_machine, App_GetFaultState());
     }
+    // Checks if the charger has thrown a fault, the disabling of the charger, etc is done with ChargeStateRunOnExit
+    else if (App_Charger_IsConnected(charger))
```

Comment:
this should be part of its own if statement. I believe the if above is designed to wait until the voltage measurements of the Module boards has settled. Charging is independent of this and therefore should be its own standalone if statement